lms/djangoapps/instructor/cybersource_enrollment_report.py

```diff
@@ -0,0 +1,169 @@
+"""
+Defines concrete class for cybersource  Enrollment Report.
+
+"""
+from courseware.access import has_access
+import collections
+from django.utils.translation import ugettext as _
+from courseware.courses import get_course_by_id
+from instructor.enrollment_report import BaseEnrollmentReportProvider
+from shoppingcart.models import RegistrationCodeRedemption, PaidCourseRegistration, CouponRedemption, OrderItem, \
+    InvoiceTransaction
+from student.models import CourseEnrollment
+
+
+class CyberSourceEnrollmentReportProvider(BaseEnrollmentReportProvider):
+    """
+    The concrete class for all CyberSource Enrollment Reports.
+    """
+
+    def get_enrollment_info(self, user, course_id):
+        """
+        Returns the User Enrollment information.
+        """
+        course = get_course_by_id(course_id, depth=0)
+        is_course_staff = has_access(user, 'staff', course)
+
+        # check the user enrollment role
+        if user.is_staff:
+            enrollment_role = _('Edx Staff')
+        elif is_course_staff:
+            enrollment_role = _('Course Staff')
+        else:
+            enrollment_role = _('Student')
+
+        course_enrollment = CourseEnrollment.get_enrollment(user=user, course_key=course_id)
+
+        if is_course_staff:
+            enrollment_source = _('Staff')
+        else:
+            # get the registration_code_redemption object if exists
+            registration_code_redemption = RegistrationCodeRedemption.registration_code_used_for_enrollment(
+                course_enrollment)
+            # get the paid_course registration item if exists
+            paid_course_reg_item = PaidCourseRegistration.get_course_item_for_user_enrollment(
```

Comment:
One thing to keep in mind, we have situations (a few times per course run), where:

user purchases course
user asks for refund
user purchases course but using a coupon

So, here, I think "paid_course_reg_item" can actually be a collection. I don't think you need to change anything in terms of code, but it's something to be aware of.